lib/script.ts

```diff
@@ -34,26 +33,21 @@ export default class Script {
 
     const evalsha = new Command("evalsha", [this.sha].concat(args), options);
     evalsha.isCustomCommand = true;
+    evalsha.promise = evalsha.promise.catch((err: Error) => {
+      if (err.toString().indexOf("NOSCRIPT") === -1) {
+        throw err;
+      }
+      const command = new Command("eval", [this.lua].concat(args), options);
```

Comment:
Replacing that with a `SCRIPT LOAD` would be another task.
Does this work in cluster mode?
```js
redis._addedScriptHashes[this.sha] = true
return redis.pipeline(['script', 'load', this.lua], ['evalsha', this.sha]])
  .exec()
  .then(([,res]) => res)
```

Reply:
I'd have other fixes that uses pipeline-based script loading: https://github.com/marcbachmann/ioredis/compare/fix-redis-reconnect-NOSCRIPT-errors...marcbachmann:pipelined-evalsha?expand=1
As it's a bit bigger, and I don't like to merge it into this pr.

The only disadvantage is that it doesn't use `script exits` checks anymore and just uses `script load` after the interval expired. I guess redis already hashes by sha internally and only loads the script if it's not present. So that should be fine.
But as there's no roundtrip anymore and script caching time can be set much higher with it. NOSCRIPT errors are also retried in there (should also work better for the cluster support).

Reply:
This is now ready in https://github.com/luin/ioredis/pull/1499
@artur-ma want to try? 😁